clang/test/dpct/cub/devicelevel/device_radix_sort.cu

```diff
@@ -1,240 +1,586 @@
 // UNSUPPORTED: cuda-8.0, cuda-9.0, cuda-9.1, cuda-9.2, cuda-10.0, cuda-10.1, cuda-10.2, cuda-11.0, cuda-11.2, cuda-11.4
 // UNSUPPORTED: v8.0, v9.0, v9.1, v9.2, v10.0, v10.1, v10.2, v11.0, v11.2, v11.4
 // RUN: dpct --format-range=none -in-root %S -out-root %T/devicelevel/device_radix_sort %S/device_radix_sort.cu --cuda-include-path="%cuda-path/include" -- -std=c++14 -x cuda --cuda-host-only
-// RUN: FileCheck --input-file %T/devicelevel/device_radix_sort/device_radix_sort.dp.cpp %s
+// RUN: FileCheck %s --match-full-lines --input-file %T/devicelevel/device_radix_sort/device_radix_sort.dp.cpp
 
 #include <cub/cub.cuh>
 
 int n, *d_keys_in, *d_keys_out, *d_values_in, *d_values_out;
+// CHECK: dpct::io_iterator_pair<int *> key_buffers(d_keys_in, d_keys_out);
+cub::DoubleBuffer<int> key_buffers(d_keys_in, d_keys_out);
+// CHECK: dpct::io_iterator_pair<int *> value_buffers(d_values_in, d_values_out);
+cub::DoubleBuffer<int> value_buffers(d_values_in, d_values_out);
+
+// CHECK: auto double_buffer(dpct::io_iterator_pair<int *> &buffers) {
+auto double_buffer(cub::DoubleBuffer<int> &buffers) {
+  // CHECK: buffers.selector = !buffers.selector;
+  buffers.selector = !buffers.selector;
+  // CHECK: buffers.iter[0] += 1;
+  buffers.d_buffers[0] += 1;
+  // CHECK: *buffers.iter[1] = 5;
+  *buffers.d_buffers[1] = 5;
+  // CHECK: buffers = dpct::io_iterator_pair<int *>(buffers.output(), buffers.input());
+  buffers = cub::DoubleBuffer<int>(buffers.Alternate(), buffers.Current());
```

Comment:
Investigate cases: calling sort within expression, calling sort in a statement of if/while/for without braces.

Reply:
Not applicable after changing API and migration of sort_pairs/sort_keys, no .swap inserted.